File: gherkin/dotnet/Gherkin.TokensGenerator/TokenFormatterBuilder.cs
using System;
using System.Linq;
using System.Text;

namespace Gherkin.TokensGenerator
{
    class TokenFormatterBuilder : IAstBuilder<object>
    {
        private readonly TestTokenFormatter formatter = new TestTokenFormatter();
        private readonly StringBuilder tokensTextBuilder = new StringBuilder();

        public string GetTokensText()
        {
            return tokensTextBuilder.ToString();
        }

        public void Build(Token token)
        {
            tokensTextBuilder.AppendLine(formatter.FormatToken(token));
        }

        public void StartRule(RuleType ruleType)
        {
            //nop
        }

        public void EndRule(RuleType ruleType)
        {
            //nop
        }

        public object GetResult()
        {
            return new object();
        }

        public void Reset()
        {
            //nop
        }

    }
}

File: gherkin/java/src/main/java/io/cucumber/gherkin/TokenFormatterBuilder.java
package io.cucumber.gherkin;

public class TokenFormatterBuilder implements Parser.Builder<String> {
    private final TokenFormatter formatter = new TokenFormatter();
    private final StringBuilder tokensTextBuilder = new StringBuilder();

    @Override
    public void build(Token token) {
        tokensTextBuilder.append(formatter.formatToken(token)).append("\n");
    }

    @Override
    public void startRule(Parser.RuleType ruleType) {
    }

    @Override
    public void endRule(Parser.RuleType ruleType) {
    }

    @Override
    public String getResult() {
        return tokensTextBuilder.toString();
    }

    @Override
    public void reset() {
    }
}
